### PR TITLE
re #1359 Query metrics /summary /summaryCounts endpoints broken

### PR DESCRIPTION
**Pull request by billoley**
use end time - 90 days as the default begin time
base time distributions on end date, not 'now'
check that begin < end, otherwise send an exception and display in the html response
change API from /summary and /summaryCounts to /summary/all (difference was response type and default date range)
change API from /summaryCounts/user to /summary/user 
use the same response object to support both calls -- just rely on the 'Accept' http header 
refactor duplicative code

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/metrics/BaseQueryMetricHandler.java [hunk 1]
```diff
@@ -79,61 +53,32 @@ public QueryMetricsSummaryHtmlResponse processQueryMetricsHtmlSummary(List<T> qu
     public void binSummary(T metric, QueryMetricsSummaryResponse summary, Date hour1, Date hour6, Date hour12, Date day1, Date day7, Date day30, Date day60,
                     Date day90) {
         Date d = metric.getCreateDate();
-        // Find out which bucket this query belongs to
+        // Find out which buckets this query belongs to based on query create date.
+        // If a query's create date is within one hour of the endDate then it is also within the other ranges.
         if (d.after(hour1)) {
             populateSummary(metric, summary.getHour1());
+        }
+        if (d.after(hour6)) {
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/metrics/BaseQueryMetricHandler.java:61]: much cleaner... thankyou

File: web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java [hunk 2]
```diff
@@ -161,29 +163,17 @@ public QueryGeometryResponse map(@PathParam("id") @Required("id") String id) {
      * @HTTP 500 internal server error
      */
     @GET
-    @Path("/summaryCounts")
```
Comment (ivakegg) [inline web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java:164]: is there anyway to at least preserve these endpoints, even if they become synonyms for you new endpoints?

File: web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java [hunk 3]
```diff
@@ -198,66 +188,40 @@ public QueryMetricsSummaryHtmlResponse getTotalQueriesSummary(@QueryParam("begin
      * @HTTP 500 internal server error
      */
     @GET
-    @Path("/summary")
```
Comment (ivakegg) [inline web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java:201]: See comment above

**Review by ivakegg — CHANGES_REQUESTED** [review]

File: web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java (hunk 2 above)
Comment (billoley) [reply web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java:164]: Added back the previous API endpoints and marked them as deprecated in the javadoc

File: web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java (hunk 3 above)
Comment (billoley) [reply web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java:201]: Added back the previous API endpoints and marked them as deprecated in the javadoc

File: web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java (hunk 2 above)
Comment (ivakegg) [reply web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsBean.java:164]: cool

**Pull request merged**